### PR TITLE
Feature : add get service instance by id

### PR DESCRIPTION
**Pull request by antechrestos**
- Add implementation to retrieve a particular service instance by its id

### CODE REVIEW

**Comment by antechrestos** [comment]
Did someone survive Christmas food? :meat_on_bone: :cake: :beer: :wine_glass: 
@nebhale : could you tell me if it is alright or if I missed something? I would need to add some functionalities such as bind service to an applicationv2, get last events... But I prefer waiting for this pull request to be resolved (one way or an other).
Thanks and enjoy new year's eve :smile:

**Comment by nebhale** [comment]
@antechrestos I'm guessing you were working on [this](https://www.pivotaltracker.com/story/show/101451546)?

File: cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/serviceinstances/ServiceInstances.java [hunk 1]
```diff
@@ -24,12 +24,19 @@
...
     Publisher<ListServiceInstancesResponse> list(ListServiceInstancesRequest request);
 
+
+    /**
+     * Makes the <a href="http://apidocs.cloudfoundry.org/214/service_instances/retrieve_a_particular_service_instance.html">Retrieve a Particular Service Instance</a> request
+     *
+     * @param request the Get Service Instance request
+     * @return the response from the Get Service Instance request
+     */
+    Publisher<ServiceInstanceResource> get(GetServiceInstanceRequest request);
```
Comment (nebhale) [inline cloudfoundry-client/src/main/java/org/cloudfoundry/client/v2/serviceinstances/ServiceInstances.java]: This should return a `GetServiceInstanceResponse`.  That type can be a straight extension of `ServiceInstanceResource` (not adding anything), but we want the abstraction in place to give us flexibility.

**Comment by nebhale** [comment]
@antechrestos Comments inline.  Quite a good contribution.

**Comment by nebhale** [comment]
@antechrestos You'll also need to rebase to `master`.  It's likely that we'll move a bit beyond whatever you eventually make it to before we merge it, but I'll take care of those last couple.  At this point you're 20-30 commit behind which is a bit too far.

**Comment by antechrestos** [comment]
@nebhale : Hi, thank you for your reply. As you asked me to, I used made some changes:
- ```GetServiceInstanceResponse``` that extends ```Resource<ServiceInstanceEntity>```: I chose not to extend ```ServiceInstanceResource``` because this late one is final and used in other response. I am opened to discuss on this point
- I made ```GetServiceInstanceRequest``` final

**Comment by nebhale** [comment]
@antechrestos I've merged the work in after making some final cosmetic changes.  Please review 530908b for the changes I've made.  They were mostly around ordering of methods (we're alphabetical) and the commit message.  If you're using IntelliJ IDEA for development, we've checked the coding style into source control, so formatting the code will automatically make all the changes I made.

Note that for next time, we'll also want you to squash the commit chain down to a single commit before merging.

**Comment by antechrestos** [comment]
@nebhale: sorry: I've just discovered the little check box __Rearrange code__  :smile:

**Comment by Zteve** [comment]
@antechrestos: That little check box stumped quite a few of us, too. Strange it isn't in the code style settings.  You might want to check “Optimize imports”, as well. Welcome.

**Pull request closed without merging**